libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -61,6 +106,9 @@ AP_ESC_Telem *AP_ESC_Telem::_singleton = nullptr;
  */
 AP_ESC_Telem *AP_ESC_Telem::get_singleton()
 {
+    if (AP_ESC_Telem::_singleton->_primary == 0) {
```

Comment:
You have to check if the singleton is nullpointer before you can dereference it here. It's not guaranteed to exist. Similarly since you selected the primary during the init, why do we need to do it again here?

Reply:
Unfortunately all ESCs initialize in push() which happens after all of the vehicle initialization has occurred

Reply:
This code is now as it was before